src/main/java/frc/robot/commands/DefaultMechCommand.java

```diff
@@ -0,0 +1,202 @@
+import frc.robot.constants.LauncherConstants.LauncherMode;
+import frc.robot.subsystems.AimingSubsystem;
+import frc.robot.subsystems.Input;
+import frc.robot.subsystems.IntakeSubsystem;
+import frc.robot.subsystems.LauncherSubsystem;
+import frc.states.MechState;
+import frc.states.mechstates.Intaken;
+import frc.states.mechstates.ReadyForAim;
+import frc.states.mechstates.ReadyForHandoff;
+import frc.states.mechstates.ReadyForIntake;
+import frc.states.mechstates.ReadyForLaunch;
+
+public class DefaultMechCommand extends Command {
+    private final IntakeSubsystem mIntakeSubsystem;
+    private final LauncherSubsystem mLauncherSubsystem;
+    private final AimingSubsystem mAimingSubsystem;
+
+    private LauncherMode mLauncherMode = LauncherMode.OFF;
+    private AimState mAimState = AimState.STOW;
+
+    private final MechState mReadyForIntake;
+    private final MechState mIntaken;
+    private final MechState mReadyForHandoff;
+    private final MechState mReadyForAim;
+    private final MechState mReadyForLaunch;
+
+    private MechState mMechState;
+
+    public DefaultMechCommand(IntakeSubsystem intakeSubsystem, LauncherSubsystem launcherSubsystem, AimingSubsystem aimingSubsystem) {
+        mIntakeSubsystem = intakeSubsystem;
+        mLauncherSubsystem = launcherSubsystem;
+        mAimingSubsystem = aimingSubsystem;
+
+        addRequirements(mIntakeSubsystem, mLauncherSubsystem, mAimingSubsystem);
+
+        mReadyForIntake = new ReadyForIntake(intakeSubsystem, launcherSubsystem);
+        mIntaken = new Intaken(launcherSubsystem);
+        mReadyForHandoff = new ReadyForHandoff(intakeSubsystem, launcherSubsystem);
+        mReadyForAim = new ReadyForAim(launcherSubsystem, aimingSubsystem);
+        mReadyForLaunch = new ReadyForLaunch(launcherSubsystem);
+
+        mMechState = determineState();
+    }
+
+    public MechState determineState() {
+        if (getIntakeBeamBreak() && getIndexerBeamBreak()) {
+            return mReadyForIntake;
+        }
+        else if (!getIntakeBeamBreak() && getIndexerBeamBreak()) {
+            return mIntaken;
+        }
+        else if (!getIntakeBeamBreak() && getIndexerBeamBreak() && mAimState == AimState.STOW) {
+            return mReadyForHandoff;
+        }
+        else if (!getIndexerBeamBreak()) {
+            return mReadyForAim;
+        }
+        else if (getIndexerBeamBreak() && isFlywheelAtSP() && isAimAtSP() && isVisionAligned() && Input.getRightBumper()) {
+            return mReadyForLaunch;
+        }
+        return null;
+    }
+
+    @Override
+    public void execute() {
+        if (Input.getX()) {
+            mMechState.setFlywheelOff();
+        }
+        else if (Input.getY()) {
+            mMechState.setSpeakerSP();
+            if (mMechState != mReadyForLaunch) {
+                mLauncherMode = LauncherMode.SPEAKER;
+            }
+            if (mMechState == mReadyForAim) {
+                mAimState = AimState.SPEAKER;
+            }
```

Comment:
This feedback applied to all the other input handling: These if statements are not necessary, the mech state itself is empty if the action cannot be completed